### PR TITLE
feat: add support for next 13

### PR DESCRIPTION
**Pull request by ZachJW34**
### User facing changelog
Support Next.js v13

### Additional details
Bring our current support for Next.js 11 + 12 to v13. No support for `turbopack` or React SC (yet).

### Steps to test
- Run `npx create-next-app@latest my-app`
- cd into app and run `npm i -D cypress`
- Check out this branch, run `yarn && yarn cypress:open`, choose `my-app`
- Walk through setup
- Add a test, I added `pages/index.cy.tsx` and run it

```ts
import Home from './index'

it('should render', () => {
  cy.mount(<Home />)
})
```

### How has the user experience changed?

https://user-images.githubusercontent.com/25158820/200433321-c23a3164-93d5-4c18-83a2-223bb74022cb.mov

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by theoludwig** [comment]
Any way we can help? Is this can be prioritized?
Thanks! :smile:

**Comment by ZachJW34** [comment]
The check for React 18 has been removed [recently](https://github.com/vercel/next.js/pull/42141), need to verify if our internal check needs to stick around. Might need it to support 13.0.0 (no longer needed in 13.0.1)

**Comment by theoludwig** [comment]
> The check for React 18 has been removed [recently](https://github.com/vercel/next.js/pull/42141), need to verify if our internal check needs to stick around. Might need it to support 13.0.0 (no longer needed in 13.0.1)

Awesome! I just tested, and indeed, I can confirm that updating Next.js (v13.0.2), works!

**Comment by astone123** [comment]
@ZachJW34 looks like everything was working fine with Next 13 except for the dependency version warning. So this PR is mainly just adding 13 in the versions list and adding a system test, right?

**Comment by ZachJW34** [comment]
@astone123 It wasn't working for Next.js v13.0.0, but v13.0.1 + didn't need any source code changes (other than the deps warning). Good to be safe, and the `hasReactRoot` property is something we missed for v12 so makes sense to keep in.

**Review by astone123 — APPROVED** [review]

File: npm/webpack-dev-server/cypress/e2e/next.cy.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /// <reference path="../support/e2e.ts" />
 import type { ProjectFixtureDir } from '@tooling/system-tests/lib/fixtureDirs'
 
-const WEBPACK_REACT: ProjectFixtureDir[] = ['next-11', 'next-12', 'next-11-webpack-4', 'next-12.1.6']
+const WEBPACK_REACT: ProjectFixtureDir[] = ['next-11', 'next-12', 'next-11-webpack-4', 'next-12.1.6', 'next-13']
```
Comment (lmiller1990) [inline npm/webpack-dev-server/cypress/e2e/next.cy.ts]: While we are here, are we able to drop next-12? Just to keep things nice and fast.We've got next-12.1.6, should cover us. Even if next-12.0.0 isn't working, we could just ask people to update to 12.1.6.

Alternatively, we could drop next-11 (two majors behind now). I am not sure if we actually landed somewhere on EOLing official support for majors, though.

Understand that the webpack resolution is different between next-12 and next-12.1.6, so if you think it's not worth the risk, that's fine.

**Review by lmiller1990 — APPROVED** [review]
Code seems fine, one comment/question - I think we need to get CI under control, it's getting slower and slower, any chance to reduce the amount of projects?

If not, all good, let's keep it in mind for other PRs.

File: npm/webpack-dev-server/cypress/e2e/next.cy.ts (hunk 1 above)
Comment (ZachJW34) [reply npm/webpack-dev-server/cypress/e2e/next.cy.ts]: I'm cool with dropping the `next-11`, at least from our system-tests.

**Pull request merged**